testdata/p4_16_samples/runtime-index-bmv2.p4

```diff
@@ -0,0 +1,95 @@
+/*
+* Copyright 2020, MNK Labs & Consulting
+* http://mnkcg.com
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*/
+
+#include <v1model.p4>
+
+typedef bit<48> mac_addr_t;
+
+header aggregator_t {
+    bit<8> val;
+}
+header vec_e_t {
+    bit<8> e;
+}
+
+header ml_hdr_t {
+    int<8> idx;
+}
+
+header ethernet_t {
+    mac_addr_t dstAddr;
+    mac_addr_t srcAddr;
+    bit<16>    etherType;
+}
+
+struct headers {
+    ethernet_t ethernet;
+    ml_hdr_t   ml;
+    vec_e_t[3] vector;
+    aggregator_t[3] pool;
+}
+
+struct metadata_t {}
+
+parser MyParser(packet_in packet, out headers hdr, inout metadata_t meta,
+                inout standard_metadata_t standard_metadata) {
+    state start {
+        packet.extract(hdr.ethernet);
+        packet.extract(hdr.ml);
+        packet.extract(hdr.vector[0]);
+        packet.extract(hdr.vector[1]);
+        packet.extract(hdr.vector[2]);
+        packet.extract(hdr.vector[0]);
```

Comment:
Did you intend to extract hdr.vector[0] twice?  I would recommend leaving this out, if you didn't intend it -- it makes the test a bit more confusing, and doesn't seem necessary at all to test the new feature.

If you do want to keep it for some reason, the packets in the STF test are shorter than the number of bytes extracted by this parser code.